### PR TITLE
Destructable platforms

### PR DESCRIPTION
**Pull request by D8H**
## Describe the example

Show how to use the "Marching Squares" extension to modify platforms dynamically.

## Checklist

- [x] I've followed all of [the best practices](http://wiki.compilgames.net/doku.php/gdevelop5/community/guide-for-submitting-an-example).
- [x] My game has a proper name in the game properties. 
- [x] My game package name behind with `com.example.`.
- [x] My game has all events unfolded.
- [x] I've added myself as the author in the game properties.
- [x] I've included a file called "README.md" with a description in proper English, explaining what this example is doing.
- [x] I confirm that this game and all of its resources can be integrated to this GitHub repository, distributed and MIT licensed.
- [x] I've cleaned unused resources.

## Game build
* https://d8h.itch.io/destructable-platformer-demo

### CODE REVIEW

**Comment by Midhil457** [comment]
Also the Desctructable platforms also doesn't convey that it has not conventional shapes as platforms. As, in a normal platformer, there are destrcutable platforms too. And also that the example is not just about desructable platforms. THought I will consider this nitpicking

I didn't find any issues with the project itself

Sorry for taking a while for the review. I completely forgot about it until I got the notification :sweat_smile:

**Comment by 4ian** [comment]
Let's go for something like "Dynamic destructible platforms"? (destructable is not proper english, and dynamic gives a hint at what the example does).

**Comment by Midhil457** [comment]
I am leaning towards `terrain` again. According to google platform means `a raised level surface on which people or things can stand` and `surface` or `raised level` is not something that comes to my mind when I see the project. Maybe like `destructible terrain platformer`?

**Comment by D8H** [comment]
> Let's go for something like "Dynamic destructible platforms"? (destructable is not proper english, and dynamic gives a hint at what the example does).

Yes, I hesitated with "dynamic" and wonder if it will be understand as it's a bit general. "dynamic" and "destructible" together feel a bit odd. Maybe "Dynamic shaped platforms" ?

**Comment by D8H** [comment]
> I am leaning towards `terrain` again. According to google, platform means `a raised level surface on which people or things can stand` and `surface` or `raised level` or `platform` is not something that comes to my mind when I see the project. Maybe like `destructible terrain platformer`?

I don't know about the common definition of the term, but "platform" is the right term for games.

**Comment by Midhil457** [comment]
When I hear `platform` I always think of area/object that are levitating in the air where the player can jump on and not the ground itself. That's why I am hesitated to use `platform` in the wording

**Review by Midhil457 — CHANGES_REQUESTED** [review]

File: examples/marching-squares-platformer/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Show how to use the "Marching Squares" extension to modify platforms dynamically.
```
[the remark this answers is not in the document]
Comment (Midhil457) [reply examples/marching-squares-platformer/README.md:1]: Haha :sweat_smile: forgot to click `finish review`. This was still marked as pending

[the remark this answers is not in the document]
Comment (D8H) [reply examples/marching-squares-platformer/README.md]: You're right I changed it to "Show how to use the "Marching Squares" extension to paint platforms and jump on it." and the example name to "Marching Squares Platforms Painter".

**Comment by D8H** [comment]
> When I hear `platform` I always think of area/object that are levitating in the air where the player can jump on and not the ground itself. That's why I am hesitated to use `platform` in the wording

They can levitate if you want them to levitate. It depends what you draw.
Moreover, the behavior to handle the character collisions is called "Platform", if the term isn't right in general, I think it works in GDevelop context.

**Comment by Midhil457** [comment]
The json's name should be the same as the folder's name. I tried fixing it and commiting, but I am not able to push the changes sadly. Great if you can do it

**Comment by D8H** [comment]
> The json's name should be the same as the folder's name. I tried fixing it and commiting, but I am not able to push the changes sadly. Great if you can do it

I renamed it.

**Comment by 4ian** [comment]
Thanks both! I think it's fine now with this name 👍

**Pull request merged**